actionpack/lib/action_dispatch/testing/integration.rb

```diff
@@ -233,7 +233,7 @@ def process(method, path, params: nil, headers: nil, env: nil, xhr: false, as: n
 
         request_env = {
           :method => method,
-          :params => request_encoder.encode_params(params),
+          :params => params.nil? ? nil : request_encoder.encode_params(params),
```

Comment:
if `encode_params` returns `nil` if `params` not present, then this conditional is not necessary

Reply:
@tsolar yeah it was the case, but i didn't commit yesterday.